### ISSUE TITLE
Navigation: Wrong cursor when an element does not have a link

### ISSUE THREAD

**Issue by hanneslsm** [WordPress/gutenberg#55672]
### Description

I have a navigation element that acts as category (Products) and has a sub menu. The element does not have a link.
I'd expect the cursor to stay `default`, but it's becoming `text`. Probably because it's set to `auto`?

### Step-by-step reproduction instructions

1. Create a navigation element without a link (eventually with sub menu)
2. watch the cursor

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/17861443/4c8633ab-129a-4d62-81ca-2cc748064df0

```
<!-- wp:navigation-link {"label":"Über uns","type":"page","id":40,"url":"http://lt.local/uber-uns/","kind":"post-type"} /-->

<!-- wp:navigation-submenu {"label":"Produkte","type":"page","id":54,"url":"","kind":"post-type"} -->
<!-- wp:navigation-link {"label":"Palettensicherung","type":"page","id":42,"url":"http://lt.local/produkte/palettensicherung/","kind":"post-type"} /-->

<!-- wp:navigation-link {"label":"Reifenverpackung","type":"page","id":25,"url":"http://lt.local/produkte/reifenverpackung/","kind":"post-type"} /-->

<!-- wp:navigation-link {"label":"Sonderbau","type":"page","id":59,"url":"http://lt.local/produkte/sonderbau/","kind":"post-type"} /-->
<!-- /wp:navigation-submenu -->

<!-- wp:navigation-link {"label":"Kontakt","type":"page","id":70,"url":"http://lt.local/kontakt/","kind":"post-type"} /-->

<!-- wp:navigation-link {"label":"English","type":"page","rel":"alternate","id":102,"url":"http://lt.local/english/","kind":"post-type","className":"prolooks-label","fontSize":"small"} /-->
```

### Environment info

- WP 6.3.2
- GB 16.9.0
- MacOS
- Brave Browser (Chromium)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by hanneslsm** [comment]
~~By the way and additionally, Lighthouse penalize me that the link is not crawl-able:~~
Ignore, setting the link to `#` solves it
<img width="792" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/52a0fb48-bdeb-4a04-b885-d20a33544be1">

**Comment by getdave** [comment]
It's now quite difficult to create Navigation items without a link. The UX optimises against it.

Also creating empty links is not a good experience for a11y or SEO.

As a result I'm going to close this out. Happy to reassess if there is a strong argument it needs to be fixed.

**Comment by hanneslsm** [comment]
> It's now quite difficult to create Navigation items without a link. The UX optimises against it.
> Happy to reassess if there is a strong argument it needs to be fixed.

I guess this is the point of the issue. Not every element in the navigation block should be necessarily a link.
It must be possible to include blocks that acts as "titles", especially for sub menu exists. This is a typical for every kind of navigation.
Example:
<pre>
- Products **(not a link)**
  - Product one (link)
  - Product two (link)
  - Product three (link)
</pre>
or
<pre>
- About **(not a link)**
  - History
  - Our team
  - Board
</pre>

Even the navigation of wordpress.org is structured this way. I think this a very strong argument 😄
https://wordpress.org/
![Image](https://github.com/user-attachments/assets/63db2cb0-4abd-4c87-bf94-e5cf708f7be3)

**Comment by getdave** [comment]
Fair point. My understanding was that it was better that those top level items should always link to a page which provides a index to access the sub pages. 

I'll reopen for now.

Appreciate your feedback.

**Comment by Anuj-Rathore24** [comment]
Hello @getdave and @hanneslsm🙋,

I have done some research and found that you can do this via adding `#` to the link. As mentioned in [this document](https://wordpress.com/support/menus/create-drop-down-menus/#add-a-menu-item-without-a-link).

On studying the codebase responsible for rendering submenus structure, there is a setting which lets submenus expand like that in wordpress.org site. But I am not able to find where we can toggle this setting in admin menus. Although the workaround mentioned above works perfectly fine!

- Codebase 
https://github.com/WordPress/gutenberg/blob/0a26fe6f727a6695c9f96f32ce3373e31441510b/packages/block-library/src/navigation-submenu/index.php#L198-L221


- Registered Toggle

https://github.com/WordPress/gutenberg/blob/0a26fe6f727a6695c9f96f32ce3373e31441510b/packages/block-library/src/navigation/edit/index.js#L661-L676

**Comment by hanneslsm** [comment]
Thanks for your input @Anuj-Rathore24 

> Although the workaround mentioned above works perfectly fine!

Yes, using a `#` creates the expected behavior of creating a link that is not clickable, however it not a good practice for a11y and SEO.

**Comment by Anuj-Rathore24** [comment]
Hey @hanneslsm, Apologies for the delay in finding the solution🙇, 

Actually the setting for creating non-clickable dropdown menus in WordPress is to edit the navigation block in Full site editing mode and toggle open on click setting as seen in the screenshot! 🥳

I can also provide a walkthrough video if you want!

![Image](https://github.com/user-attachments/assets/34525044-d458-419a-a5be-76b54a6f69e1)

**Comment by karmatosed** [comment]
I would add that having just text in navigation is even more important for mega menus and longer form. So I would +1 that having this ability is key. I am going to move a few labels on this to see if we can move the issue on it's journey. I do see it as enhancement not a bug, but that's totally fine, it is good to work in and very useful.
